dotCMS/src/main/java/com/dotcms/vanityurl/model/CachedVanityUrl.java

```diff
@@ -172,13 +99,86 @@ private String addOptionalForwardSlashSupport(final String uri){
     /**
      * This takes the uir that was originally stored in the contentlet adds validates it.
      * @param uri the uri stored in the contentlet.
-     * @param cache404VanityUrl whether or not this is a 404 cache entry
      * @return normalized uri.
      */
-    private String normalize(final String uri, final boolean cache404VanityUrl){
-        final String uriRegEx = cache404VanityUrl ? uri : addOptionalForwardSlashSupport(uri);
+    private String normalize(final String uri){
+        final String uriRegEx = addOptionalForwardSlashSupport(uri);
         return VanityUrlUtil.isValidRegex(uriRegEx) ? uriRegEx : StringPool.BLANK;
     }
+    
+    
+    public VanityUrlResult handle(final String uriIn,
+                    final HttpServletResponse response) {
+        
+        final Tuple2<String,String> rewritten = processForward(uriIn);
+        final String rewrite = rewritten._1;
+        final String queryString = rewritten._2;
+
+
+        if (this.response==301 || this.response==302 ) {
+            response.setStatus(this.response);
+            response.setHeader("Location", rewrite);
+            return new VanityUrlResult(rewrite, queryString, true);
+        }
+        
+        if (this.response==200 && UtilMethods.isSet(rewrite) && rewrite.contains("//")) {
+            Try.run(()-> new CircuitBreakerUrl(rewrite).doOut(response)).onFailure(e->{throw new DotRuntimeException(e);});
+            return new VanityUrlResult(rewrite, queryString, true);
+        }
+
+        return new VanityUrlResult(rewrite, queryString, false);
+        
+        
+    }
+    
+    
+
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((forwardTo == null) ? 0 : forwardTo.hashCode());
+        result = prime * result + (int) (languageId ^ (languageId >>> 32));
+        result = prime * result + response;
+        result = prime * result + ((siteId == null) ? 0 : siteId.hashCode());
+        result = prime * result + ((url == null) ? 0 : url.hashCode());
+        return result;
+    }
+
+
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'obj' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)